src/fsharp/CheckExpressions.fs

```diff
@@ -9366,7 +9392,18 @@ and TcNormalizedBinding declKind (cenv: cenv) env tpenv overallTy safeThisValOpt
 
         let argAndRetAttribs = ArgAndRetAttribs(argAttribs, retAttribs)
 
-        if HasFSharpAttribute cenv.g cenv.g.attrib_DefaultValueAttribute valAttribs then
+        let isZeroMethod =
+            match declKind, pat with
+            | ModuleOrMemberBinding, SynPat.Named(id, _, _, _) when id.idText = "Zero" -> 
```

Comment:
This is part of the rule for the builder's Zero and DefaultValueAttribute. - but, it looks like it could apply on any member with the name "Zero" with any kind of enclosing type?

In my mind, this rule should only apply if it's in the context of a builder. However, I don't think we have enough information to know that the member is part of a builder at this point; which might be why the check is the way it is.

I'm not sure of a better alternative, but we should put a comment here describing why it is the way it is.
We could also have additional checks to only apply the rule if "Zero" is an instance method with no parameters.

Another suggestion, but might be overkill, if we detect the zero method, we could show a warning talking about it, then it's up to the user to do a `#nowarn` for this special circumstance.

Reply:
Maybe even easier, you could just not throw the error if the attribute is used on a non-field member.